### PR TITLE
Added support for flags

### PR DESCRIPTION
**Pull request by dharmit**
So far, you use `wego 1 Paris` to fetch weather data for Paris for 1
day. With this commit, you use `wego -days 1 -city Paris` to fetch the
same information. Benefit of this approach is that (1) it prints help
message when you do `wego -h` (2) it is clear what data you're seeking.

PS - I really loved wego. And I am learning golang. So I felt like adding
this simple support for flags as a part of learning and contributing
back to wego. :smiley:

### CODE REVIEW

File: we.go [hunk 1]
```diff
@@ -482,6 +486,8 @@ func printDay(w weather) (ret []string) {
 }
 
 func init() {
+	flag.IntVar(&days, "days", 1, "Number of days of weather data to be displayed. This should be between 0 to 5")
+	flag.StringVar(&city, "city", "Paris", "City to be queried")
```
Comment (schachmat) [inline we.go]: The default should stay New York.

File: we.go [hunk 2]
```diff
@@ -482,6 +486,8 @@ func printDay(w weather) (ret []string) {
 }
 
 func init() {
+	flag.IntVar(&days, "days", 1, "Number of days of weather data to be displayed. This should be between 0 to 5")
```
Comment (schachmat) [inline we.go]: Please remove the "0 to 5" sentence, since this restriction is backend specific and might change without notice.

File: we.go [hunk 3]
```diff
@@ -562,33 +568,24 @@ func marshalLang(rv map[string]interface{}, r *resp) error {
...
 	}
 	params = append(params, "key="+config.APIKey)
 
-	for _, arg := range os.Args[1:] {
-		if v, err := strconv.Atoi(arg); err == nil && len(arg) == 1 {
-			numdays = v
-		} else {
-			config.City = arg
-		}
-	}
```
Comment (schachmat) [inline we.go]: I want to retain the feature to specify an alternate location or number of days without using flags (for brevity). Just replace the loop through os.Args with a loop through flag.Args(). Flags should take precedence over non-flag arguments, so `wego -days 1 -city Paris 4 'New York'` should return one day of forecast for paris. However `wego 4 'New York'` should still return 4 days of forecast for New York.

File: we.go [hunk 4]
```diff
@@ -562,33 +568,24 @@ func marshalLang(rv map[string]interface{}, r *resp) error {
...
-	params = append(params, "num_of_days="+strconv.Itoa(numdays))
+	params = append(params, "num_of_days="+strconv.Itoa(days))
 	params = append(params, "tp=3")
 	if config.Lang != "" {
 		params = append(params, "lang="+config.Lang)
 	}
 
-	// fmt.Fprintln(os.Stderr, params)
-
 	res, err := http.Get(uri + strings.Join(params, "&"))
```
Comment (schachmat) [inline we.go]: Please don't remove those comments. If you want, add a 'debug' flag to enable them in another pull request.

File: we.go [hunk 5]
```diff
@@ -599,8 +596,6 @@ func main() {
 		log.Fatal(err)
 	}
 
-	// fmt.Println(string(body))
-
```
Comment (schachmat) [inline we.go]: dito

**Comment by schachmat** [comment]
Heyho @dharmit,

since you are learning go, i left you the small things, I would have fixed myself as an exercise. ;) If you don't want to implement them yourself, I will do it. If you are new to git as well:

1. Rebase your flag-support branch onto the new upstream master (I just merged another PR).
2. Fix the things I commented on in the diff.
3. `git commit -a --amend` will fixup your previous commit.
4. `git push -f` the branch.

**Comment by dharmit** [comment]
Hey @schachmat,

Thanks so much for the feedback on the PR. I will make the suggested changes and resubmit the PR as soon as I can. :)

File: we.go (hunk 3 above)
Comment (dharmit) [inline we.go]: Hi @schachmat, below is the code:
~~~
    flag.Parse()
    config.City = city

    for _, arg := range flag.Args() {
                if v, err := strconv.Atoi(arg); err == nil && len(arg) == 1 { 
                        days = v 
                } else {
                        config.City = arg 
                }
        }
~~~

When I run it as `wego -days 1 -city Mumbai 2 Paris`, it returns 2 days' data for Paris and not 1 day data for Mumbai. So `flag.Args()`, which returns non-flag arguments, is replacing the values set earlier. I don't see a straightforward way of doing this other than using `if...else`. But before I go that way, I want to be sure that I'm thinking correct.

Comment (schachmat) [inline we.go]: That is correct. The condition has been used before and is necessary to differentiate between the  two possible arguments, when no flags are used.

**Comment by schachmat** [comment]
Heyho @dharmit,

do you have the time to update your PR in the next few days, or should I do it?

**Comment by dharmit** [comment]
Hi @schachmat , Yes, I will work on this later today.

**Comment by dharmit** [comment]
Hi @schachmat , I have modified the PR. Let me know your feedback.

**Comment by schachmat** [comment]
Merged. I cleaned up some variables (merged them into the Config struct) and decided, that giving non-flag arguments priority is way easier to implement, so I included that change as well.

Thanks for your contribution @dharmit.

**Comment by dharmit** [comment]
@schachmat 
> giving non-flag arguments priority is way easier to implement

Indeed. The `break` approach I took, looked bad to me.

> Thanks for your contribution @dharmit.

Pleasure was all mine. :smile:

**Pull request closed without merging**